index.js

```diff
@@ -240,36 +240,40 @@ FSWatcher.prototype._awaitWriteFinish = function(path, threshold, callback) {
   (function awaitWriteFinish (prevStat) {
     fs.stat(fullPath, function(err, curStat) {
       if (err) {
-        delete this._pendingWrites[path];
-        if (err.code == 'ENOENT') return;
-        return callback(err);
+        // prevent `unlink` race condition
+        this._getWatchedDir(sysPath.dirname(path))
+          .remove(sysPath.basename(path));
+        if (this._pendingWrites[path]) this._pendingWrites[path].cancelWait();
+        if (err.code !== 'ENOENT') callback(err);
+        return;
```

Comment:
If the file is deleted (err.code === 'ENOENT') callback will never be called.

[Here](https://github.com/alexpusch/chokidar/commit/e7c33696bd246adb39b341f80c80cf2859692a76) is my try for this fix. I've extracted the creation of _pendingWrites[path] outside of the polling function, and returned the task of deleting the entry to the unlink event (via cancelWait). [everything is green](https://travis-ci.org/alexpusch/chokidar)